app/controllers/admin/admin_users_controller.rb

```diff
@@ -144,11 +147,43 @@ def update_user
 
         @user.save
         flash[:notice] = success_message
-        redirect_to request.referer || root_path
+        if params[:admin_action] == 'spamban'
+          redirect_to confirm_delete_user_creations_admin_user_path(@user)
+        else
+          redirect_to request.referer || root_path
+        end
 
       end
     end
   end
+  
+  before_filter :user_is_banned, only: [:confirm_delete_user_creations, :destroy_user_creations]  
+  def user_is_banned
+    @user = User.find_by_login(params[:id])
+    unless @user && @user.banned?
+      flash[:error] = ts("That user is not banned!")
+      redirect_to admin_users_path and return
+    end
+  end
+  
+  def confirm_delete_user_creations
+    @works = @user.works.paginate(page: params[:works_page])
+    @comments = @user.comments.paginate(page: params[:comments_page])
+    @bookmarks = @user.bookmarks
```

Comment:
It's only displaying the counts on them, the ActiveRecord relation shouldn't actually evaluate anything else?